test/e2e/declarative_test_fwk/model.py

```diff
@@ -146,13 +146,18 @@ def __repr__(self) -> str:
         return str(self)
 
 
+# fields for steps associated with a resource
+class ResourceStepDict(TypedDict, total=False):
+    resource: ResourceDict
+    steps: List[StepDict]
+
+
 # fields in a test scenario
 class ScenarioDict(TypedDict, total=False):
     id: str
     description: str
     marks: List[str]
-    resource: ResourceDict
-    steps: List[StepDict]
```

Comment:
Resource can be specified at each step as create, patch, delete allow specifying ResourceDict attributes at each step.